narwhal/primary/src/metrics.rs

```diff
@@ -537,7 +539,14 @@ impl PrimaryMetrics {
             .unwrap(),
             current_round: register_int_gauge_vec_with_registry!(
                 "current_round",
-                "Current round the node is in",
+                "Current round the node will propose",
+                &["epoch"],
+                registry
+            )
+            .unwrap(),
+            highest_processed_round: register_int_gauge_vec_with_registry!(
```

Comment:
@mwtian I am not opposed adding this metric, but just want to understand what's the difference in the level of information we'll get compared to the `current_round` metric. Given the fact that we are immediately [forwarding a valid certificate to proposer](https://github.com/MystenLabs/sui/blob/cce65069bceee8a5863dcfb6b608ca0c47908916/narwhal/primary/src/core.rs#L414) the `proposer` to "reset" its round, proposer will create a header for the `current_round` which basically will only differ by 1 compared to the `highest_processed_round`. I see a case where those two can diverge if we constantly receive higher round certificates without giving any opportunity to the proposer to received a formed quorum of parents - but not sure whether we do have any suspicions for that. All in all, no problem adding this, but just want to make sure we don't have overlap in semantics

Reply:
I agree in normal operations this should be very close to `current_round`. I'm mostly interested in the case of node restarting and catching up. If this metric is shown to be not very useful, we can remove it later.